flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/runner/ZooKeeperDefaultDispatcherRunnerTest.java

```diff
@@ -153,15 +154,17 @@ void testResourceCleanupUnderLeadershipChange() throws Exception {
         final TestingLeaderElectionService dispatcherLeaderElectionService =
                 new TestingLeaderElectionService();
 
-        final CuratorFramework client =
-                ZooKeeperUtils.startCuratorFramework(configuration, fatalErrorHandler)
-                        .asCuratorFramework();
-        try (final TestingHighAvailabilityServices highAvailabilityServices =
-                new TestingHighAvailabilityServicesBuilder()
-                        .setDispatcherLeaderElectionService(dispatcherLeaderElectionService)
-                        .setJobMasterLeaderRetrieverFunction(
-                                jobId -> ZooKeeperUtils.createLeaderRetrievalService(client))
-                        .build()) {
+        try (final CuratorFrameworkWithUnhandledErrorListener curatorFrameworkWrapper =
+                        ZooKeeperUtils.startCuratorFramework(configuration, fatalErrorHandler);
```

Comment:
The following way is not used here as it will pop up a warning: `CuratorFrameworkWithUnhandledErrorListener used without 'try'-with-resources statement`.

```
try (CuratorFramework curatorFramework =
                        ZooKeeperUtils.startCuratorFramework(configuration, fatalErrorHandler)
                                .asCuratorFramework();
```